urlextract.py

```diff
@@ -71,17 +71,18 @@ class URLExtract:
         ("`", "`"),
     }
 
-    def __init__(self):
+    def __init__(self, cache_dir=os.path.dirname(__file__)):
```

Comment:
Could you assign `cache_dir=None` instead and call `os.path.dirname(__file__)` inside the `__init__`? I would like to avoid evaluating of default argument. Even if this is not the problematic case. 
Thank you.